lib/auth-providers/cmd.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Fetches a bearer token via comamnd
+ */
+
+'use strict';
+
+// for API compatability
+/* eslint no-sync: 0 */
+const spawnSync = require('child_process').spawnSync;
+
+module.exports = {
+  refresh: function (config) {
+    return new Promise((resolv, reject) => {
+      const cmd  = config['cmd-path'];
+      const args = config['cmd-args'].split(' ');
+
+      const output = spawnSync(cmd, args, {
+        windowsHide: true
+      });
+
+      let result;
+      try {
+        result = JSON.parse(output.stdout.toString('utf8'));
+      } catch (err) {
+        return reject('Failed to run cmd.');
```

Comment:
can we `reject` with an `Error` instance instead of the `string`?

Reply:
Oops, yeah will do.